FeedStoreChallenge/RealmFeedStore.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  RealmFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Jorge Lucena on 10/1/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+public final class RealmFeedStore {
+
+	private let configuration: Realm.Configuration
+
+	private var _realm: Realm?
+
+	private let queue = DispatchQueue(label: "\(RealmFeedStore.self)Queue", qos: .userInitiated)
+	
+	public init(storeURL: URL) {
+		configuration = Realm.Configuration(fileURL: storeURL)
+	}
+
+	private func getRealm() throws -> Realm {
+		try _realm ?? Realm(configuration: configuration, queue: queue)
+	}
+}
+
+extension RealmFeedStore: FeedStore {
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		queue.async { [weak self] in
+			do {
+				guard let realm = try self?.getRealm() else {
+					return completion(nil)
+				}
```

Comment:
There's no need to weakify `self` in this context - so you don't need to deal with optionals.

Moreover, if it cannot get the Realm instance, it shouldn't complete with `nil` - it should complete with an error since the `delete operation` failed.

Capturing a strong reference to `self` will solve this problem since there won't be an `Optional` to unwrap.

Or even better, you don't even need to capture `self` - for example:

```suggestion
		queue.async { [configuration, queue] in
			do {
				let realm = try Realm(configuration: configuration, queue: queue)
```

Reply:
Good advice, thanks!